tests/approx_matching_ut.cpp

```diff
@@ -274,23 +267,29 @@ SIMPLE_UNIT_TEST_SUITE(ApproxMatchingTest) {
 		APPROXIMATE_SCANNER(fsm, 2) {
 			ACCEPTS(text);
 
-			for (size_t posLeft = 0; posLeft < text.size() / 2; ++posLeft) {
+			for (size_t posLeft = 0; posLeft < text.size() / 2 - 1; ++posLeft) {
 				size_t posRight = text.size() - posLeft - 1;
-				for (int operation = 0; operation < 3; ++operation) {
-					auto changedText = ChangeText(text, operation, posLeft, posRight);
-					ACCEPTS(changedText);
+				for (int operationLeft = MutateOperation::Begin; operationLeft < MutateOperation::End; ++operationLeft) {
+					for (int operationRight  = MutateOperation::Begin; operationRight < MutateOperation::End; ++operationRight) {
+						auto changedText = ChangeText(text, operationRight, posRight);
+						changedText = ChangeText(changedText, operationLeft, posLeft);
+						ACCEPTS(changedText);
+					}
 				}
 			}
 		}
 
 		APPROXIMATE_SCANNER(fsm, 1) {
 			ACCEPTS(text);
 
-			for (size_t posLeft = 0; posLeft < text.size() / 2; ++posLeft) {
+			for (size_t posLeft = 0; posLeft < text.size() / 2 - 1; ++posLeft) {
```

Comment:
IMHO
`- 1` in `for (...; posLeft < text.size() / 2 - 1; ...)` looks unexpected.
It would be great to add a comment explaining its purpose.